source/lib.civet

```diff
@@ -3440,35 +3404,55 @@ function typeOfJSXFragment(node, config, getRef) {
  * uses await, or just adding async if specified.
  * Returns an Array suitable for `children`.
  */
-function wrapIIFE(exp, async) {
-  let prefix, suffix
-  if (async) {
-    prefix = "(async ()=>"
-    suffix = ")()"
-  } else if (hasAwait(exp)) {
-    prefix = "(await (async ()=>"
-    suffix = ")())"
-  } else {
-    prefix = "(()=>"
-    suffix = ")()"
-  }
+function wrapIIFE(expressions: StatementTuple[], async): ASTNode[] {
+  let prefix
 
-  // TODO: This rest prevents an infinite recursion bug, ideally it wouldn't be necessary
-  const expressions = Array.isArray(exp) ? [...exp] : [exp]
-  const block = {
+  if async
+    async = "async "
+  else if hasAwait expressions
+    async = "async "
+    prefix =
+      type: "Await"
+      children: ["await "]
+
+  block := {
     type: "BlockStatement",
     expressions,
     children: ["{", expressions, "}"],
     bare: false,
   }
-  updateParentPointers(block)
 
-  // TODO: This should return a call expression ideally
-  return [
-    prefix,
-    block,
-    suffix,
-  ]
+  parameters :=
+    type: "Parameters"
+    children: ["()"]
+    names: []
+
+  signature :=
+    modifier:
+      async: !!async
+    returnType: undefined
+
+  fn := {
+    type: "ArrowFunction",
+    signature
+    parameters
+    returnType: undefined
+    ts: false
+    async
+    block
+    children: [async, parameters, "=>", block]
+  }
+
+  updateParentPointers(block, fn)
+
+  exp :=
+    type: "CallExpression"
+    children: [ makeLeftHandSideExpression(fn), "()" ]
+
+  if prefix
+    return [ makeLeftHandSideExpression [prefix, exp] ]
+
+  return [ exp ]
```

Comment:
Do we still need the array wrappers? I'm guessing we can change the return type to `ASTNode` (but haven't tried).

Reply:
It could probably be `ASTNode` or `StatementTuple[]` as long as it is the same type that gets passed to `insertReturn`. I'm leaning towards keeping more things as `StatementTuple[]` in preparation for statementizing more things soon.

Reply:
Oh, for this specific place we could either use the array wrapper up update all the use locations to wrap in a `children` array.